### PR TITLE
stdenv, cacert: $NIX_SSL_CERT_FILE changes

### PR DESCRIPTION
**Pull request by vcunat**
###### Motivation for this change

Some SSL libs don't react to `$SSL_CERT_FILE`.  That actually makes sense to me, as we add this behavior as nixpkgs-specific, so it seems "safer" to use `$NIX_*`.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date: _docs are **still** missing ;-)_
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/stdenv/generic/setup.sh [hunk 1]
```diff
@@ -646,9 +646,12 @@ fi
...
 
 
-# Prevent OpenSSL-based applications from using certificates in
-# /etc/ssl.
+# Prevent SSL libraries from using certificates in /etc/ssl, unless set explicitly.
 # Leave it in shells for convenience.
+if [ -z "${NIX_SSL_CERT_FILE:-}" ] && [ -z "${IN_NIX_SHELL:-}" ]; then
+  export NIX_SSL_CERT_FILE=/no-cert-file.crt
+fi
+# Another variant left for compatibility.
```
Comment (layus) [inline pkgs/stdenv/generic/setup.sh]: I think it would make sense to enable this in pure shells, for consistency. But that would be for another PR.

**Review by layus — APPROVED** [review]
It totally makes sense, and should have been done before :-)

File: pkgs/stdenv/generic/setup.sh (hunk 1 above)
Comment (vcunat) [reply pkgs/stdenv/generic/setup.sh]: :heart:  see the new commit.

**Comment by LnL7** [comment]
My reasoning back when I added it was to go trough the upstream codepath instead of our patches by default. But that doesn't really make sense since we want NIX_SSL_CERT_FILE to work for everything.
